php_serialize/unserialize.go

```diff
@@ -53,120 +52,180 @@ func (us *UnSerializer) Decode() (PhpValue, error) {
 	if token, _, err := us.r.ReadRune(); err == nil {
 		switch token {
 		default:
-			us.saveError(fmt.Errorf("php_serialize: Unknown token %#U", token))
+			return nil, fmt.Errorf("php_serialize: Unknown token %#U", token)
 		case TOKEN_NULL:
-			value = us.decodeNull()
+			value, err = us.decodeNull()
+			if err != nil {
+				return nil, err
+			}
```

Comment:
```suggestion
			return us.decodeNull()
```
Will that work? The code would be much shorter if it does

Reply:
fixed